src/Libraries/Microsoft.Extensions.Caching.Hybrid/Microsoft.Extensions.Caching.Hybrid.csproj

```diff
@@ -20,12 +20,15 @@
     <!-- This package needs to reference the dotnet9 versions of Microsoft.Extensions packages as it depends on
     surface area that was added in .NET 9. -->
     <ForceLatestDotnetVersions>true</ForceLatestDotnetVersions>
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+    <UseLoggingGenerator>true</UseLoggingGenerator>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.Caching.Abstractions" />
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" />
     <PackageReference Include="Microsoft.Extensions.Options" />
+    <ProjectReference Include="..\Microsoft.Extensions.Telemetry.Abstractions\Microsoft.Extensions.Telemetry.Abstractions.csproj" PrivateAssets="All" />
```

Comment:
The build mechanism here has a custom generator for `[LoggerMessage]` etc (https://github.com/dotnet/extensions/tree/main/src/Generators/Microsoft.Gen.Logging); a limitation of that custom tooling is that it doesn't work at all if it can't resolve the types it expects for optional features that we aren't using . A longer-term goal might be to fix the tooling, but for now this is a pragmatic build fix